package/MDAnalysis/lib/mdamath.py

```diff
@@ -92,7 +94,39 @@ def normal(vec1, vec2):
     n = norm(normal)
     if n == 0.0:
         return normal  # returns [0,0,0] instead of [nan,nan,nan]
-    return normal / n  # ... could also use numpy.nan_to_num(normal/norm(normal))
+    # ... could also use numpy.nan_to_num(normal/norm(normal))
+    return normal / n
+
+
+def pdot(a, b):
+    """Pairwise dot product.
+
+    ``a`` must be the same shape as ``b``.
+
+    Parameters
+    ----------
+    a: :class:`numpy.ndarray` of shape (N, M)
+    b: :class:`numpy.ndarray` of shape (N, M)
+
+    Returns
+    -------
+    :class:`numpy.ndarray` of shape (N,)
+    """
+    return np.einsum('ij,ij->i', a, b)
+
+
+def pnorm(a):
+    """Euclidean norm of each vector in a matrix
+
+    Parameters
+    ----------
+    a: :class:`numpy.ndarray` of shape (N, M)
+
+    Returns
+    -------
+    :class:`numpy.ndarray` of shape (N,)
+    """
+    return pdot(a, a)**0.5
```

Comment:
I was just curious if `np.sqrt` is any different
```
In [5]: x = np.random.random((5000,))

In [6]: %timeit x**0.5
11.5 µs ± 60.6 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [7]: %timeit np.sqrt(x)
11.5 µs ± 67.7 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```
but it gives the same time on my laptop.

Reply:
The performance differs for bigger numbers.

![Screenshot 2020-06-06 at 6 40 52 AM](https://user-images.githubusercontent.com/31115101/83920976-b9bdbb00-a7c0-11ea-853e-3bb1c774fef4.png)

Had to quietly take that out from my "Intro to numpy and why it's faster than everything else" tutorial once.

Reply:
I am easily swayed by data – thanks!